### PR TITLE
Fix: delete the secret of the cluster

### LINKED ISSUE #5467: [Bug]  can't delete the cluster when endpoint of cluster is wrong or changed
**Describe the bug**

If the IP of the cluster is wrong when adding the cluster, it will not be possible to delete it from the apiserver.

**To Reproduce**

1、The config I used.
```
apiVersion: v1
clusters:
- cluster:
    insecure-skip-tls-verify: true
    server: https://36.11.17.167:6443
  name: member1
contexts:
- context:
    cluster: member1
    user: kubernetes-admin
  name: kubernetes-admin@member1
current-context: kubernetes-admin@member1
kind: Config
preferences: {}
users:
- name: kubernetes-admin
  user:
    client-certificate-data: LS0tLS1CRUdJTiBDRVJUSUrZJQ0FURS0tLS0tCk1JSURJekNDQWd1Z0F3SUJBZ0lJVSt4RVNESHM4aFF3RFFZSktvWklodmNOQVFFTEJRQXdGVEVUTUJFR0ExVUUKQXhNS2EzVmlaWEp1WlhSbGN6QWdGdzB5TWpFeE1ETXdOelV6TWpCYUdBOHlNVEl5TVRBeE1UQTRNREF5TmxvdwpOREVYTUJVR0ExVUVDaE1PYzNsemRHVnRPbTFoYzNSbGNuTXhHVEFYQmdOVkJBTVRFR3QxWW1WeWJtVjBaWE10CllXUnRhVzR3Z2dFaU1BMEdDU3FHU0liM0RRRUJBUVVBQTRJQkR3QXdnZ0VLQW9JQkFRQzl3RGxsbTZZRzE5azgKM3hzTkFXQ2ZhYkpHeWd0TExRTmhHOVpxb2tSNTlHTFhjUzZ3UnBuOThjczV1a3hZS09Qd1pHNXMydWkyN3ovMwo5R3FydWJ5cStXMXBQUzZBd3NxSkZHcVh1NUUxN1BZclNkaVFXMVphaFNIZXlxY1crd0ZpbnM3SHp6aUtQb3lSClVNbzZuL0tTZVR2QWFFZXo0NUxDSkZNSHJ0eUFSUUhRdUpremZHUU9UOXM0MndsQ3A5OUpYdkFVV0ZreVo0UWQKaXRhMzJKWGNjQzR1MndjMXBMS2lSTmlWdWhWVVcyTkhGbHhraGxQNHhWWlE2WkZLcksvLzc3bkxqc2ZNcWQ3cwpjMldFdStGZUNsSGlRMGRpdExWd2tvWDlVWjlUMWVsVzhjL2ZpTVRQRlBKRzdMYUh6czg5OG5tRVpaN3BGa0JGCkhvRTRVMTYvQWdNQkFBR2pWakJVTUE0R0ExVWREd0VCL3dRRUF3SUZvREFUQmdOVkhTVUVEREFLQmdnckJnRUYKQlFjREFqQU1CZ05WSFJNQkFmOEVBakFBTUI4R0ExVWRJd1FZTUJhQUZEMDNpdFNLUzlEaytiNTNjMEI1TUpoUgpZZm1BTUEwR0NTcUdTSWIzRFFFQkN3VUFBNElCQVFBTkpQd0tFeTBaRjVPT1RHM2dsWHlrVndUSnkrVlVxdlRwCmhiV2JicmRUSko1SE9SNzQ1UlU3bm9OV3F4dmZOY3k4NjVwWEc5RFB6WXN6OVRaMWpTd2NwOTRZdWEydzZyOUoKVGRwN2llNm8xLy9acSswVEhFRGhiTzRiMU12RzExSm16Zyt4bUp3WXhpRlh3c3VOam9IVDRqNUd6a2ROTFFkUQoyM3k4Y3FCVEkyM2YrbTFYMFI5MjFBcWZBL0szY3RwaEY4aVZFbi9yVlcxdEFIQVhRTWxqWC9va2I0Y2R3ZXpnCldRWmtOZUMzQnFMSW14aTFFaGN4TXdiVmJBSmFjM0lRbWNnNUhFWFNVNTJIV0dwQldldWVxaHlwZ2dnaWlMOWYKL0pBbnRVMmtheCtHY01PamNHZHQyVDVNQmFRbTM5WXgxUnVRU2JiYWRybUhrVExQWUtIRwotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==
    client-key-data: LS0tLS1CRUdJTiBSU0EgUFJJVdkFURSBLRVktLS0tLQpNSUlFcEFJQkFBS0NBUUVBdmNBNVpadW1CdGZaUE44YkRRRmduMm15UnNvTFN5MERZUnZXYXFKRWVmUmkxM0V1CnNFYVovZkhMT2JwTVdDamo4R1J1Yk5yb3R1OC85L1JxcTdtOHF2bHRhVDB1Z01MS2lSUnFsN3VSTmV6MkswblkKa0Z0V1dvVWgzc3FuRnZzQllwN094ODg0aWo2TWtWREtPcC95a25rN3dHaEhzK09Td2lSVEI2N2NnRVVCMExpWgpNM3hrRGsvYk9Oc0pRcWZmU1Y3d0ZGaFpNbWVFSFlyV3Q5aVYzSEF1THRzSE5hU3lva1RZbGJvVlZGdGpSeFpjClpJWlQrTVZXVU9tUlNxeXYvKys1eTQ3SHpLbmU3SE5saEx2aFhncFI0a05IWXJTMWNKS0YvVkdmVTlYcFZ2SFAKMzRqRXp4VHlSdXkyaDg3UFBmSjVoR1dlNlJaQVJSNkJPRk5ldndJREFRQUJBb0lCQUhIZmhCS0lRRERuSklXMQpWcVNOYVJFQisyMGtIM2IwQW8wcWlaamczTHRiZ0YzT25MaEpCbTB4eUVaU21USm5CN1BpY2FwK0RSNkxvaFlKCjVtSThTdzlJTlBMdE5wVXJVU1RHemRsVlJTWFptVG9OVFJ3RTFHUHFkWk1naHZueGxMazhtTmVzY2ovMG5hTXYKbXMxbjQ4a25YRVp4em1ZVWNxUno3Q2cyTmpuN0p0empYbTlkWmtSMXo4aktJSFk3ZmFIT3JWRXd2MFNCQWxrKwpvN0ExYkRndUZiclg4V3R6ckNnaG9ocFBFWUFVZzlzbTZld2hBMFp0cnNPN00vaVA0Rno0QmNEaW5PSWdIczFiCmcwbnpvNzFsY2NnQ2xDNndqL2xtb3NscnUzR3NMbEVXQlRIYS9rbjlYdEY3aEh5ZnhjdkF2K1ZYbHdMeXhhd3MKcXc0NmNZRUNnWUVBMFFFTDdEZzRlWkY4bWpNdWZ5NWxRRFM5b084eHlKSC9uMGs5QWJac1VHUFM1V2J3b2hqcgp2Y3JUN0l3OTlzTkcxTXc5NzBnVzhaSC9qVEM3ZEk3K1Bzam5nWDQ0M1NkbEVEWk1KaGgzK1hpWERLdll5OCs4CldTZzlOZHdVN29CU1d5alhFTS9wYkJJcHdIRGhaZCtocFZvMHpSMjZ6TUg5NzdaRkR2TGpJbDhDZ1lFQTZHcm0KZDUyZ3BWelczeFZjZndtWlNKRXlwZjFKK0pmMzVOM2cvcnhrM0g4RWJVcGZJNTlQL3BFOGNkbVl5VkVwWEIxOAo2bE11MU96R01jZ1pwbGsydjQxVzIzYm1YY3hvVStPOWtDQVQyN1JqK3NDNi9DOG42VDRFM0RkN1JxRDY4YXM0CkFsTjcvUmoxWlVQUGExRW1LRDFKM2Z4NEp3QTVkN1hNRlBPMjM2RUNnWUVBeHZCMUZkTHB4cVY1OGhqWG9FMWMKS1FKS29aNHNSOXJZY3hpMlRpUUM4MzRoSnlTVjFlVmN0eG9TSUszWTZVSTA4S2E4S1QvNFB0YVdaaUZGdlF2Qwo2NlkxM2lkMEh1VzVwL1hYQThmMG91eHE4cSs4T3p1djd6bURFZm83bmsxY2x0TEg2NFBMam9lM1lDclNFVE1OClhTUUE4R0RsQnJOOGI2aVNFVW1XYzRjQ2dZQVhkbk05c0JQL25aZXpmWnlRMW84NjNDMll4REVyZjJ2dkZ0Y1gKNER6dVpZUTA1RVVtZTNhLzVEbU41TUdmMFNtenV1WlpZWHhnMG1aUHNEdW1wMnIzZTNWQlFvZVBBSXVuckYzKwpwcjVhZDZPQU5PU3JnTGN1dkt3UTd5anF3R1dVOEF2L2NCNGNOdURQRUVGb1BCRDJEdm5lQnhhNkdUT3orZzVZCkx5SlBBUUtCZ1FDVmNKQ2JqOE5aUjBiWDJRZHpmeVR5VlJ2dW1SdVp1ZkR4ekxua2ZZVGhmSnA5NmcvTnAwRGIKOExISU5QWUlNeHhkMGFLUmR3N3J2VndhcHVZUmY1aDhPUllsSjZoUjdBYlFSY2J3VXp5ZDU1eDcycGlUVXA5bwp3dldzYUQzaWpNS1IxK0lUTHg5ZFNPczNVdXlSN0p1S014UUFOYm1Cc2kzU3hvekJtaDFIVkE9PQotLS0tLUVORCBSU0EgUFJJVkFURSBLRVktLS0tLQo=
```
2、Other operations I have done
image
3、See error
image

**Expected behavior**

delete the cluster successfully from the apiserver. The reason is because when adding a cluster, even if the IP is wrong, the secret will be created, but the configMap will not be created. When displaying the cluster list, the secret will be queried, and when the cluster is deleted, only the configmap will be searched.

**Screenshots**

**KubeVela Version**

1.7

**Cluster information**

1.21
**Additional context**

### PR DESCRIPTION
**Pull request by suwliang3**
Signed-off-by: suwanliang_yewu <suwanliang_yewu@cmss.chinamobile.com>

### Description of your changes

Fixes #5467 

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

**Review by wonderflow — CHANGES_REQUESTED** [review]
please add some test case for it

File: pkg/apiserver/domain/service/cluster.go [hunk 1]
```diff
@@ -387,7 +387,11 @@ func (c *clusterServiceImpl) DeleteKubeCluster(ctx context.Context, clusterName
 	cluster, err := c.getClusterFromDataStore(ctx, clusterName)
 	if err != nil {
 		if errors.Is(err, datastore.ErrRecordNotExist) {
-			return nil, bcode.ErrClusterNotFoundInDataStore
+			if err = multicluster.DetachCluster(ctx, c.K8sClient, clusterName); err != nil {
```
Comment (Somefive) [inline pkg/apiserver/domain/service/cluster.go:390]: The fallback code here should follow through to the logic below.

**Comment by nuclearwu** [comment]
Please handle go/lint and go/ check-diff error

**Review by Somefive — APPROVED** [review]
Great job!

**Review by wonderflow — APPROVED** [review]
good job

**Review by FogDong — APPROVED** [review]
Should this be backported to 1.7?

**Pull request merged**